core/crypto/vrf/vrfhash_provider.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_CRYPTO_VRF_VRF_HASH_PROVIDER_HPP
+#define CPP_FILECOIN_CORE_CRYPTO_VRF_VRF_HASH_PROVIDER_HPP
+
+#include "crypto/randomness/randomness_types.hpp"
+#include "crypto/vrf/vrf_types.hpp"
+#include "primitives/address/address.hpp"
+
+namespace fc::crypto::vrf {
+  /**
+   * @class VRFHashProvider used to create VRFHash according to lotus hash implementation:
+   * https://github.com/filecoin-project/lotus/blob/1914412adf3c81028fcc305b887ca8ad189bc2dc/chain/gen/gen.go#L579
+   */
+  class VRFHashProvider {
```

Comment:
Is a class really needed here if all it has is one static method? Why not a free function?

Reply:
refactored